photutils/isophote/isophote.py

```diff
@@ -740,34 +771,71 @@ def _isophote_list_to_table(isophote_list):
             `~photutils.isophote.IsophoteList` instance
         A list of isophotes.
 
+    key_properties : A list of properties to export from the isophote_list
+        If key_properties = ['all'] or ['main'], it will pick all or few
+        of the main properties.
+
     Returns
     -------
     result : `~astropy.table.QTable`
-        An astropy QTable with the main isophote parameters.
+        An astropy QTable with the selected or all isophote parameters.
     """
 
     properties = OrderedDict()
-    properties['sma'] = 'sma'
-    properties['intens'] = 'intens'
-    properties['int_err'] = 'intens_err'
-    properties['eps'] = 'ellipticity'
-    properties['ellip_err'] = 'ellipticity_err'
-    properties['pa'] = 'pa'
-    properties['pa_err'] = 'pa_err'
-    properties['grad'] = 'grad'
-    properties['grad_error'] = 'grad_err'
-    properties['grad_r_error'] = 'grad_rerr'
-    properties['x0'] = 'x0'
-    properties['x0_err'] = 'x0_err'
-    properties['y0'] = 'y0'
-    properties['y0_err'] = 'y0_err'
-    properties['ndata'] = 'ndata'
-    properties['nflag'] = 'flag'
-    properties['niter'] = 'niter'
-    properties['stop_code'] = 'stop_code'
-
     isotable = QTable()
 
+    # main_properties: `List`
+    # A list of main parameters matching the original names of
+    # the isophote_list parameters
+
+    def __rename_properties(properties,
```

Comment:
Also, the renamed output columns here introduces a backwards-incompatible breaking change.  Given the potential headaches for users (and for consistency), it's probably better to also just rename the property attributes at the same time.  The old attribute names can be kept with deprecation warnings.  But I can do that in a separate PR -- justing pointing that out for my own reference.

Reply:
Yes, I wasn't sure what was behind the changed  names, but I kept the original as is, so that I minimize the number of my changes.